r2i/prediction.cc

```diff
@@ -0,0 +1,159 @@
+    error.Set (RuntimeError::Code::MEMORY_ERROR,
+               "Triying to access an non-existing index");
+    return 0;
+  }
+
+  float *fdata = static_cast<float *>(this->GetResultData(output_index, error));
+  if (RuntimeError::Code::EOK != error.GetCode()) {
+    return 0;
+  }
+  if (nullptr == fdata) {
+    error.Set (RuntimeError::Code::INVALID_FRAMEWORK_PARAMETER,
+               "Prediction result not set yet");
+    return 0;
+  }
+
+  return fdata[index];
+}
+
+void *Prediction::GetResultData (unsigned int output_index,
+                                 RuntimeError &error) {
+  if (output_index >= this->results_data.size()) {
+    error.Set(RuntimeError::Code::INVALID_FRAMEWORK_PARAMETER,
+              "Output index out of bounds");
+    return 0;
+  }
+
+  return this->results_data[output_index].get();
+}
+
+unsigned int Prediction::GetResultSize (unsigned int output_index,
+                                        RuntimeError &error) {
+  if (output_index >= this->results_sizes.size()) {
+    error.Set(RuntimeError::Code::INVALID_FRAMEWORK_PARAMETER,
+              "Output index out of bounds");
+    return 0;
+  }
+
+  return this->results_sizes[output_index];
+}
+
+RuntimeError Prediction::AddResult (float *data, unsigned int size) {
+  RuntimeError error;
+
+  if (nullptr == data) {
+    error.Set (RuntimeError::Code::NULL_PARAMETER,
+               "Invalid tensor values passed to prediction");
+    return error;
+  }
+
+  if (0 == size) {
+    error.Set (RuntimeError::Code::NULL_PARAMETER,
+               "Invalid tensor size passed to prediction");
+    return error;
+  }
+
+  float *internal_data = (float *)malloc(size * sizeof(float));
+  if (nullptr == internal_data) {
+    error.Set(RuntimeError::Code::MEMORY_ERROR,
+              "Error while allocating prediction result memory.");
+    return error;
+  }
+
+
+  void *result = memcpy(internal_data, data, size * sizeof(float));
+  if (nullptr == result) {
+    error.Set(RuntimeError::Code::MEMORY_ERROR,
+              "Error while copy memory to prediction result");
+    return error;
+  }
+
+  this->results_data.push_back(std::shared_ptr<float>(internal_data, free));
+  this->results_sizes.push_back(size);
+
+  return error;
+}
+
+RuntimeError Prediction::InsertResult(unsigned int output_index, float *data,
+                                      unsigned int size) {
+  RuntimeError error;
+
+  if (output_index >= this->results_data.size()) {
+    error.Set(RuntimeError::Code::INVALID_FRAMEWORK_PARAMETER,
+              "Output index out of bounds");
+    return error;
+  }
+
+  if (nullptr == data) {
+    error.Set (RuntimeError::Code::NULL_PARAMETER,
+               "Invalid tensor values passed to prediction");
+    return error;
+  }
+
+  if (0 == size) {
+    error.Set (RuntimeError::Code::NULL_PARAMETER,
+               "Invalid tensor size passed to prediction");
+    return error;
+  }
+
+  float *internal_data = this->results_data[output_index].get();
```

Comment:
verify current output index result size if big enough to contain the new data

Reply:
I've reverted this change.